[Bug]: Missing dependencies in `sysdep-install` script

### Summary

I'm trying to build this from source on a very fresh install of Manjaro Linux running KDE Plasma, but I get a number of issues during installation which seem to indicate that not all dependencies are installed by the `sysdep-install` script.


### Steps to Reproduce

Install from source using the below instructions from the README:

> Install script from source:
> 
> ```
> npm install # Installs dependencies for building
> npm run sysdep-install # Install system dependencies for building
> npm run bi-install # Installs all Bismuth components
> ```

The `sysdep-install` script on line 2 runs `scripts/sysdep-install.sh`. For Manjaro, this just installs `cmake`, `ninja`, and `extra-cmake-modules`.

```
...
    "arch" | "manjaro")
      sudo pacman -S --noconfirm --needed \
        cmake ninja extra-cmake-modules
      ;;
...
```

However, when I run the `bi-install` script on line 3, I get an error complaining that a C compiler was not found:

```
❯ npm run bi-install

> bismuth@2.2.0 prebi-install
> npm run build


> bismuth@2.2.0 build
> scripts/build.sh

🏗️ Building Bismuth...
CMake Error at CMakeLists.txt:6 (project):
  No CMAKE_C_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.


CMake Error at CMakeLists.txt:6 (project):
  No CMAKE_CXX_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
```

This seems to indicate that `sysdep-install` did not install a C compiler. This error can be fixed by installing the `gcc` package from pacman (or "Add/Remove Software"). However, a new error then appears:

```
❯ npm run bi-install

> bismuth@2.2.0 prebi-install
> npm run build


> bismuth@2.2.0 build
> scripts/build.sh

🏗️ Building Bismuth...
-- The C compiler identification is GNU 11.1.0
-- The CXX compiler identification is GNU 11.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Installing in /usr/local. Run /home/murchu27/bin/bismuth/build/prefix.sh to set the environment for bismuth.
-- Configuring done
CMake Error at src/kcm/CMakeLists.txt:8 (add_library):
  The install of the kcm_bismuth target requires changing an RPATH from the
  build tree, but this is not supported with the Ninja generator unless on an
  ELF-based or XCOFF-based platform.  The CMAKE_BUILD_WITH_INSTALL_RPATH
  variable may be set to avoid this relinking step.
```

Resolving this had nothing to do with `CMAKE_BUILD_WITH_INSTALL_RPATH`; I just had to delete the CMake cache, and then run the installer again:

```
❯ rm build/CMakeCache.txt
❯ npm run bi-install
# installation completes successfully...
🎉 Installation finished.
💡 You can enable and configure window tiling in the System Settings > Window Management > Window Tiling.
🦾 Don't forget to enable window tiling. We hope you will enjoy it!
```

### Expected behavior

`sysdep-install` should also install a C compiler. This would look something like:

```
...
    "arch" | "manjaro")
      sudo pacman -S --noconfirm --needed \
        gcc cmake ninja extra-cmake-modules
      ;;
...
```

Then `bi-install` should run successfully first time on Arch-based distros like Manjaro

### Screenshots

_No response_

### Bismuth version

`master` branch source

### KDE Plasma version

5.23.4

### The platform KWin is running on

X11

### Additional context

_No response_

--- a/scripts/sysdep-install.sh
+++ b/scripts/sysdep-install.sh
@@ -14,7 +14,7 @@ if [ -f /etc/os-release ]; then
 
     "ubuntu" | "pop" | "debian" | "neon" )
       sudo apt-get install -y \
-        cmake ninja-build extra-cmake-modules kirigami2-dev \
+        g++ cmake ninja-build extra-cmake-modules kirigami2-dev \
         libkf5config-dev libkf5configwidgets-dev libkf5coreaddons-dev \
         libkf5declarative-dev libkf5i18n-dev libkf5kcmutils-dev \
         libqt5svg5-dev qml-module-qtquick* qtbase5-dev \
@@ -38,7 +38,7 @@ if [ -f /etc/os-release ]; then
 
     "arch" | "manjaro")
       sudo pacman -S --noconfirm --needed \
-        cmake ninja extra-cmake-modules
+        gcc cmake ninja extra-cmake-modules
       ;;
 
     "void")

